aibolit/metrics/cc/main.py

```diff
@@ -1,80 +1,86 @@
-            tmppath = os.path.dirname(os.path.realpath(__file__))
-            shutil.copyfile(os.path.join(tmppath, 'pom.xml'), root + '/pom.xml')
-            shutil.copyfile(os.path.join(tmppath, 'cyclical.xml'), root + '/cyclical.xml')
-            if showoutput:
-                subprocess.run(['mvn', 'pmd:pmd'], cwd=root, check=False)
-            else:
-                subprocess.run(['mvn', 'pmd:pmd'], cwd=root,
-                               stdout=subprocess.DEVNULL,
-                               stderr=subprocess.DEVNULL, check=False)
-            if os.path.isfile(root + '/target/pmd.xml'):
-                res = self.__parseFile(root)
-                return res
-            raise Exception('File ' + self.input + ' analyze failed')
-        finally:
-            shutil.rmtree(root)
-
-    def __parseFile(self, root):
-        result = {'data': [], 'errors': []}
-        content = []
-        with open(root + '/target/pmd.xml', 'r', encoding='utf-8') as file:
-            content = file.read()
-            soup = BeautifulSoup(content, 'lxml')
-            files = soup.find_all('file')
-            for file in files:
-                out = file.violation.string
-                name = file['name']
-                pos1 = name.find(root + '/src/main/java/')
-                pos1 = pos1 + len(root + '/src/main/java/')
-                name = name[pos1:]
-                pos1 = out.find('has a total cyclomatic complexity')
-                pos1 = out.find('of ', pos1)
-                pos1 = pos1 + 3
-                pos2 = out.find('(', pos1)
-                complexity = int(out[pos1:pos2 - 1])
-                result['data'].append({'file': name, 'complexity': complexity})
-            errors = soup.find_all('error')
-            for error in errors:
-                name = error['filename']
-                pos1 = name.find(root + '/src/main/java/')
-                pos1 = pos1 + len(root + '/src/main/java/')
-                name = name[pos1:]
-                raise Exception(error['msg'])
-        return result
+from aibolit.ast_framework import AST, ASTNode, ASTNodeType
+
+
+class CCMetric:
+    """
+    Calculates cyclomatic complexity by counting
+    decision points in all methods and constructors:
+
+    - Base complexity: 1 per method/constructor
+    - Decision points: if, while, for, do-while,
+                       switch cases, catch, throw,
+                       ternary, break, continue
+    - Boolean operators (&&, ||) add additional complexity paths
+    """
+
+    def value(self, ast: AST) -> int:
+        total_complexity = 0
+
+        for method in ast.get_proxy_nodes(ASTNodeType.METHOD_DECLARATION):
+            total_complexity += self._method_complexity(ast, method)
+
+        for constructor in ast.get_proxy_nodes(ASTNodeType.CONSTRUCTOR_DECLARATION):
+            total_complexity += self._method_complexity(ast, constructor)
+
+        return total_complexity
```

Comment:
I wonder if we can refactor it to this.
Does it make sense? Or does it make the code shorter, but not more clear?

```suggestion
        return sum(
            self._method_complexity(ast, node)
            for node in ast.get_proxy_nodes(
                ASTNodeType.METHOD_DECLARATION,
                ASTNodeType.CONSTRUCTOR_DECLARATION),
            )
        )
```